include/alpaka/core/UniformCudaHip.hpp

```diff
@@ -0,0 +1,133 @@
+/* Copyright 2019 Axel Huebl, Benjamin Worpitz, Matthias Werner, René Widera
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+
+#include <alpaka/core/BoostPredef.hpp>
+
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !BOOST_LANG_CUDA
+    #error If ALPAKA_ACC_GPU_CUDA_ENABLED is set, the compiler has to support CUDA!
+#endif
+
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED) && !BOOST_LANG_HIP
+    #error If ALPAKA_ACC_GPU_HIP_ENABLED is set, the compiler has to support HIP!
+#endif
+
+// Backend specific includes.
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+    #include <alpaka/core/Cuda.hpp>
+#else
+    #include <alpaka/core/Hip.hpp>
+#endif
+
+#include <string>
+
+
+namespace alpaka
+{
+    namespace uniform_cuda_hip
+    {
+        namespace detail
+        {
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+            using Error_t = cudaError;
+#else
+            using Error_t = hipError_t;
+#endif
+            //-----------------------------------------------------------------------------
+            //! CUDA runtime API error checking with log and exception, ignoring specific error values
+            ALPAKA_FN_HOST inline auto rtCheck(
+                Error_t const & error,
+                char const * desc,
+                char const * file,
+                int const & line)
+            -> void
```

Comment:
(Not important), can't we skip `-> void` now?

Reply:
yes we can, but no need to do now.